pennylane/plugins/tests/conftest.py

```diff
@@ -0,0 +1,136 @@
+# limitations under the License.
+"""Contains shared fixtures for the device tests."""
+import os
+import pytest
+import numpy as np
+import pennylane as qml
+
+# ==========================================================
+# pytest fixtures
+
+# seed for random functions
+np.random.seed(42)
+# Tolerance for analytic tests
+TOL = 1e-6
+# Tolerance for non-analytic tests
+TOL_STOCHASTIC = 0.05
+# Number of shots to call the devices with
+N_SHOTS = 50000
+# List of all devices that are included in PennyLane
+LIST_CORE_DEVICES = {"default.gaussian", "default.qubit", "default.qubit.tf"}
+
+
+@pytest.fixture(scope="function")
+def tol():
+    """Numerical tolerance for equality tests. Returns a different tolerance for tests
+    probing analytic or non-analytic devices (yielding deterministic or stochastic test results)."""
+
+    def _tol(analytic):
+        if analytic:
+            return float(os.environ.get("TOL", TOL))
+        else:
+            return TOL_STOCHASTIC
+
+    return _tol
+
+
+@pytest.fixture(scope="session")
+def init_state():
+    """Fixture to create an n-qubit random initial state vector."""
+
+    def _init_state(n):
+        state = np.random.random([2 ** n]) + np.random.random([2 ** n]) * 1j
+        state /= np.linalg.norm(state)
+        return state
+
+    return _init_state
+
+
+@pytest.fixture(scope="session")
+def skip_if():
+    """Fixture to skip tests."""
+
+    def _skip_if(condition):
+        if condition:
+            pytest.skip("Test does not apply to this device.")
+
+    return _skip_if
+
+
+@pytest.fixture(scope="function")
+def device(device_kwargs):
+    """Fixture to create a device."""
+
+    def _device(n_wires):
+        device_kwargs["wires"] = n_wires
+        return qml.device(**device_kwargs)
+
+    return _device
+
+
+# ============================
+# These functions are required to define the device name to run the tests for
+
+
+def pytest_addoption(parser):
+    """Add command line option to pytest."""
+
+    # TODO: find a way to pass an unknown number of optional command line options to pytest
+    # so we can optionally provide compulsory args like "cutoff_dim" or specific args such as a backend
+
+    parser.addoption("--device", action="store", default=None)
+    parser.addoption("--shots", action="store", default=None, type=int)
+    parser.addoption("--analytic", action="store", default=None)
+
+
+def pytest_generate_tests(metafunc):
+    """Set up fixtures from command line options. """
+    opt = metafunc.config.option
+
+    device_kwargs = {
+        "name": opt.device,
+        "shots": opt.shots,
+        "analytic": opt.analytic,
+    }
+
+    # =======================
+    # some processing of the command line options
+    if device_kwargs["shots"] is None:
+        device_kwargs["shots"] = N_SHOTS
+    if device_kwargs["analytic"] is None:
+        # use default value of device
+        device_kwargs.pop("analytic")
+    else:
+        # turn string into boolean

```

Comment:
Not sure if there is a more elegant way, but we cannot use ``type=bool``, because we need a third option ``None`` for if the flag is unused and the default value should be used...

Reply:
Ah 🤔 

In the meantime, might be good to use `device_kwargs["analytic"].lower()` to allow if the user passes `'false'`, to avoid it accidentally becoming True.